core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -842,10 +842,34 @@ private[spark] class Master(
   }
 }
 
+object SparkUrl {
+
+  def unapply(sparkUrl: String): Option[(String, Int)] = {
```

Comment:
actually - rather than introducing a new class and do pattern matching, can you just add a function to the Utils class that return an Option[(String, Int)]?